c7n/resources/rdscluster.py

```diff
@@ -216,5 +211,141 @@ def process(self, snapshots):
+                client.restore_db_cluster_from_snapshot(**params)
+                self.create_db_instances_within_cluster(latest_snapshot)
+            except ClientError as e:
+                raise
+
+    """
+      Right now im only returning the first value returned by the cli. I sort them from bottom to top then append the top
+      value to the array then return the array.  I am setting this function up to handle an array of different
+      cluster's snapshot, but for right now I just want to get the single cluster restore working then Ill focus on
+      creating multiple rds cluster instances that have same values within an array in the policy.
+    """
+    @staticmethod
+    def get_latest_snapshot(snapshots):
+
+        x = []
+        prev_obj = None
+
+        for snapshot in snapshots:
+
+            curr_obj = snapshot
+
+            if prev_obj is None:
+                prev_obj = curr_obj
+                x.append(curr_obj)
+
+            if curr_obj['DBClusterIdentifier'] == prev_obj['DBClusterIdentifier']:
+                prev_obj = curr_obj
+            else:
+                x.append(curr_obj)
+                prev_obj = curr_obj
+
+        print x
+
+        return x
+
+    def create_db_instances_within_cluster(self, snapshot):
+
+        db_instances = self.data.get('db-instances') or {}
+        num_of_instances = db_instances.get('number', 1)
+        db_identifier = db_instances.get('identifier', self.get_default_identifier(snapshot['DBClusterIdentifier']))
+        db_size = db_instances.get('size', 'db.r3.large')
+        post_fix = db_instances.get('post-fix', 'dev')
+
+        if db_identifier is None:
+            raise ClientError('MUST PROVIDE A db_identifier IN YOUR POLICY: Please delete cluster before trying again')
+        if db_size is None:
+            self.log_default_value('size:: default size is :: %s , which is minimum size', db_size)
+        if num_of_instances is None:
+            self.log_default_value('number:: number of db instances which is: %s', num_of_instances)
+
+        for instance in xrange(0, num_of_instances):
+
+            if instance == 0:
+                new_db_identifier = db_identifier + '-' + post_fix
+            else:
+                new_db_identifier = db_identifier + '-' + post_fix + '-' + str(instance)
+
+            params = {'DBClusterIdentifier': snapshot['DBClusterIdentifier'],
+                      'Engine': snapshot['Engine'],
+                      'DBInstanceClass': db_size,
+                      'DBInstanceIdentifier': new_db_identifier,
+                      }
+
+            client = local_session(self.manager.session_factory).client('rds')
+
+            try:
+                client.create_db_instance(**params)
+            except ClientError as e:
+                raise
+
+    """
+    TODO: Need to delete cluster when failure on database instance creation.  Wondering how I can call the Delete class
+    from within this function
+    """
```

Comment:
Yeah i was planning on reusing the existing rds cluster delete, however, I havent figured out how to call it from my Restore class.